### PR TITLE
feat(logging): redact PII from all bot log output

### PR DESCRIPTION
**Pull request by Lazare-42**
## What

Always-on PII redaction for every log surface this bot uploads to S3 (bot.log / logs.log, sound.log, speaker_separation.log, browser-console relay), so logs can be shared with on-prem customers.

- New `src/utils/PiiRedactor.ts`: static pattern passes (meeting URLs incl. Teams URL-encoded context and `teams.microsoft.com/meet/…` consumer links, generic/SSO URLs, ws/rtmp stream URLs with credentials, JWTs/Bearer/secret-ish KEY=VALUE, AWS keys, emails incl. `%40`, phones with a ≥7-digit guard, IPv4) + a dynamic dictionary of observed speaker names and the bot name mapped to stable placeholders (`<SPEAKER_n>`, `<BOT_NAME>`) so logs stay correlatable.
- Wired at the winston format choke point (covers all console output incl. page-console relay), plus direct file writers (speaker_separation.log, sound.log) and the chat result log line (`<CHAT_TEXT>`).
- Fail-closed: any internal redaction error yields `<REDACTION_FAILED>`, never the raw line. Speaker registry capped at 500. Escape hatch: `DISABLE_LOG_PII_REDACTION=true`.
- Deliberately NOT redacted (product artifacts consumed by the pipeline): diarization.jsonl, chat_messages.json. Open policy decision, flagged in the edge-case CSV.
- bot_uuid, unix timestamps, file paths and metrics are asserted to survive (negative tests).

## Tests

`src/utils/__tests__/PiiRedactor.test.ts` runs every applicable row of the vendored 53-row edge-case CSV (source of truth lives in the meeting-baas-v2 monorepo) plus behavior tests: 40/40 green. Patterns were validated against two real production log bundles (a 219k-line Teams log and a Meet log): zero PII leaks, no false positives on metrics after the phone-pattern fix.

Pre-existing (untouched by this PR): 18 urlParser/meet.test failures also present on the base commit.

### CODE REVIEW

**Comment by Lazare-42** [comment]
@coderabbitai review

File: src/proxy/toggle-proxy.ts [hunk 1]
```diff
@@ -47,16 +47,60 @@ let server: Server | null = null
...
+}
+
 /** Country code + IANA timezone of the current exit IP, or null until probed. */
 export function getExitGeo(): { country: string | null; timezone: string | null } | null {
   return exitGeo
 }
 
+export function markProxyDisabledReason(reason: string): void {
+  proxyDisabledReason = reason
+}
```
[the remark this answers is not in the document]
Comment (Lazare-42) [reply src/proxy/toggle-proxy.ts:76]: Out of scope for this PR — `toggle-proxy.ts` comes from the proxy-telemetry commit this branch is stacked on, not the PII change. Leaving for a follow-up.

**Comment by Lazare-42** [comment]
@coderabbitai review

File: src/proxy/toggle-proxy.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lazare-42) [reply src/proxy/toggle-proxy.ts:76]: @coderabbitai yes, please open the follow-up issue.

**Review by amr-meetingbaas — APPROVED** [review]
